packages/wonder-blocks-button/components/button.js

```diff
@@ -134,25 +134,35 @@ export type SharedProps = {|
      * href is not
      */
     onClick?: (e: SyntheticEvent<>) => mixed,
-
-    /**
-     * Run async code before navigating. If the promise returned rejects then
-     * navigation will not occur.
-     *
-     * If both safeWithNav and beforeNav are provided, beforeNav will be run
-     * first and safeWithNav will only be run if beforeNav does not reject.
-     */
-    beforeNav?: () => Promise<mixed>,
-
-    /**
-     * Run async code in the background while client-side navigating. If the
-     * navigation is server-side, the callback must be settled before the
-     * navigation will occur. Errors are ignored so that navigation is
-     * guaranteed to succeed.
-     */
-    safeWithNav?: () => Promise<mixed>,
 |};
 
+type Props =
+    | {|
+          ...SharedProps,
+      |}
+    | {|
+          ...SharedProps,
+
+          href: string,
+
+          /**
+           * Run async code before navigating. If the promise returned rejects then
+           * navigation will not occur.
+           *
+           * If both safeWithNav and beforeNav are provided, beforeNav will be run
+           * first and safeWithNav will only be run if beforeNav does not reject.
+           */
+          beforeNav?: () => Promise<mixed>,
+
+          /**
+           * Run async code in the background while client-side navigating. If the
+           * navigation is server-side, the callback must be settled before the
+           * navigation will occur. Errors are ignored so that navigation is
+           * guaranteed to succeed.
+           */
+          safeWithNav?: () => Promise<mixed>,
+      |};
+
```

Comment:
Thanks for doing this. Be prepared for webapp issues though if folks spread these props for some reason. Our version of flow in webapp is not good to spread unions.

Reply:
Thanks for the heads up.